comp/core/autodiscovery/listeners/snmp_deduper.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+package listeners
```

Comment:
This could benefit from being it's own lib/package, separated from the listener so that not everything is accessible and we can focus on the API between the two regardless of the implementation